test/extended/machines/scale.go

```diff
@@ -186,6 +187,35 @@ func scaleMachineSet(name string, replicas int) error {
 }
 
 var _ = g.Describe("[sig-cluster-lifecycle][Feature:Machines][Serial] Managed cluster should", func() {
+
+	var (
+		c      *kubernetes.Clientset
+		dc     dynamic.Interface
+		helper *bmhelper.BaremetalTestHelper
+	)
+
+	g.BeforeEach(func() {
```

Comment:
IMHO, instead wrapping up with BeforeEach and AfterEach just for Baremetal, I think it would be more convenient to use if baremetal and defer statements. Because in reality, this change has no effect on other platforms.

Reply:
BeforeEach/AfterEach are the ways supported by [ginkgo](https://onsi.github.io/ginkgo/) for structuring the setup/teardown in a clear/readable mannder, and also ensuring the proper order of execution. I'd find complicated to follow other custom approaches for that (the helper teardown will not have any effect in case of non-baremetal platform)